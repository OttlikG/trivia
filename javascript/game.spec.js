var Game = require('./game').Game;
var Random = require('./prng').Random;

describe("The test environment", function() {
  it("should pass", function() {
    expect(true).toBe(true);
  });

  it("should access game", function() {
    expect(Game).toBeDefined();
  });
});

describe("Game", function () {
  let consoleSpy
  let _console = console.log
  let _random = Math.random

  beforeEach(() => {
    consoleSpy = jest.fn()
    console.log = consoleSpy
  })

  afterEach(() => {
    console.log = _console
    Math.random = _random
  })

  it('Should be the same output on every run', () => {
    Math.random = jest.fn(() => 0.5)
    
    const game = new Game()
    game.add('Chet')
    game.add('Pat')

    game.run()

    expect(consoleSpy.mock.calls.flat()).toMatchSnapshot()
  })

  test.each([
    [0],
    [1],
    [2],
    [3],
  ])('Should be the same output on every run', (seed) => {
    let random = new Random(seed)
    Math.random = jest.fn(() => random.nextFloat())

    const game = new Game()
    game.add('Player1')
    game.add('Player2')
    game.add('Player3')
    game.add('Player4')

    game.run()

    expect(consoleSpy.mock.calls.flat()).toMatchSnapshot()
  })
});
