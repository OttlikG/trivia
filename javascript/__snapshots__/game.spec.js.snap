// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Should be the same output on every run 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Chet was added",
    ],
    Array [
      "They are player number 1",
    ],
    Array [
      "Pat was added",
    ],
    Array [
      "They are player number 2",
    ],
    Array [
      "Chet is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Chet's new location is 4",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 0",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Chet now has 1 Gold Coins.",
    ],
    Array [
      "Pat is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Pat's new location is 4",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 1",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Pat now has 1 Gold Coins.",
    ],
    Array [
      "Chet is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Chet's new location is 8",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 2",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Chet now has 2 Gold Coins.",
    ],
    Array [
      "Pat is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Pat's new location is 8",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 3",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Pat now has 2 Gold Coins.",
    ],
    Array [
      "Chet is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Chet's new location is 0",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 4",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Chet now has 3 Gold Coins.",
    ],
    Array [
      "Pat is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Pat's new location is 0",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 5",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Pat now has 3 Gold Coins.",
    ],
    Array [
      "Chet is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Chet's new location is 4",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 6",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Chet now has 4 Gold Coins.",
    ],
    Array [
      "Pat is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Pat's new location is 4",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 7",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Pat now has 4 Gold Coins.",
    ],
    Array [
      "Chet is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Chet's new location is 8",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 8",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Chet now has 5 Gold Coins.",
    ],
    Array [
      "Pat is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Pat's new location is 8",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 9",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Pat now has 5 Gold Coins.",
    ],
    Array [
      "Chet is the current player",
    ],
    Array [
      "They have rolled a 4",
    ],
    Array [
      "Chet's new location is 0",
    ],
    Array [
      "The category is Pop",
    ],
    Array [
      "Pop Question 10",
    ],
    Array [
      "Answer was correct!!!!",
    ],
    Array [
      "Chet now has 6 Gold Coins.",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
